### DISCUSSION TITLE
[Feature Request] OnHover in DropDown

### DISCUSSION THREAD

**Discussion by devappsms** [heroui-inc/heroui#1244] [Feature requests]
### Is your feature request related to a problem? Please describe.

Is there any possibility to add hover as option on MenuTriggerType in Dropdown. or do we have any work around for this. 

### Describe the solution you'd like

Dropdown to show the menus/options while hover over the Dropdown

### Describe alternatives you've considered

NA

**Comment by m-naeem66622** [comment]
I think there should include an option in dropdown to view options on hover too instead of clicking it. These types of dropdowns will be helpful in navbar to show different categories by hovering on it rather than clicking.

**Comment by goustaff** [comment]
Oops, I made a similar post. :x 
https://github.com/nextui-org/nextui/discussions/1806

**Comment by timomedia** [comment]
I also hope to add this option to the dropdown

**Reply by cairumeng** [reply]
I use Tooltip and Listbox, it seems work fine for me

**Reply by rwchampin** [reply]
This medium article shows how to toggle the Dropdown menu on hover.  [Toggle NextUI Dropdown Menu on Hover](https://medium.com/@rwchampin/nextui-dropdown-on-hover-open-close-nextui-dropdown-component-on-hover-9733b1572480)

**Comment by rwchampin** [comment]
This medium article shows how to toggle the Dropdown menu on hover.  [Toggle NextUI Dropdown Menu on Hover](https://medium.com/@rwchampin/nextui-dropdown-on-hover-open-close-nextui-dropdown-component-on-hover-9733b1572480)

**Comment by doseeing** [comment]
Toggle dropdown solution based on [Toggle NextUI Dropdown Menu on Hover](https://medium.com/@rwchampin/nextui-dropdown-on-hover-open-close-nextui-dropdown-component-on-hover-9733b1572480), fix menu disappear when move cursor from dropdown trigger to dropdown menu.

```typescript
"use client";

import {
  DropdownItem,
  DropdownTrigger,
  Dropdown,
  DropdownMenu,
  Avatar,
  DropdownSection,
} from "@nextui-org/react";

import { useState } from "react";

export const HoverDropdown = () => {
  const [isOpen, setIsOpen] = useState(false);
  const [timeoutId, setTimeoutId] = useState<any>(null);
  const delay = 1000;
  return (
    <Dropdown isOpen={isOpen}>
      <DropdownTrigger>
        <Button
          onMouseEnter={() => {
            clearTimeout(timeoutId);
            setIsOpen(true);
          }}
          onMouseLeave={() => {
            const id = setTimeout(() => setIsOpen(false), delay);
            setTimeoutId(id);
          }}
        >Hover Me</Button> 
      </DropdownTrigger>
      <DropdownMenu
        aria-label="Profile Actions"
        variant="flat"
        onMouseEnter={() => {
          clearTimeout(timeoutId);
          setIsOpen(true);
        }}
        onMouseLeave={() => {
          setIsOpen(false);
        }}
      >
          <DropdownItem key="profile" className="h-14 gap-2">
            <p className="font-semibold">Signed in as</p>
            <p className="font-semibold">zoey@example.com</p>
          </DropdownItem>
          <DropdownItem key="settings">My Settings</DropdownItem>
          <DropdownItem key="team_settings">Team Settings</DropdownItem>
          <DropdownItem key="analytics">Analytics</DropdownItem>
          <DropdownItem key="system">System</DropdownItem>
          <DropdownItem key="configurations">Configurations</DropdownItem>
          <DropdownItem key="help_and_feedback">Help & Feedback</DropdownItem>
          <DropdownItem key="logout" color="danger">
            Log Out
          </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
};
```

**Comment by embium** [comment]
Anyone have any ideas on how to make this a multilevel dropdown menu?

**Comment by serhatkaya** [comment]
```typescript
export const AccountDropdown: FC<CommonProps> = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div onMouseEnter={() => setIsOpen(true)} onMouseLeave={() => setIsOpen(false)}>
      <Dropdown isOpen={isOpen}>
        <DropdownTrigger>
          <Button
            variant="light"
            size="sm"
            startContent={<IconUserFilled />}
            onMouseEnter={() => setIsOpen(true)}
          >
            Sign In
          </Button>
        </DropdownTrigger>
        <DropdownMenu aria-label="Static Actions">
          <DropdownItem key="new">New file</DropdownItem>
          <DropdownItem key="copy">Copy link</DropdownItem>
          <DropdownItem key="edit">Edit file</DropdownItem>
          <DropdownItem key="delete" className="text-danger" color="danger">
            Delete file
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>
    </div>
  );
};
```

Works fine

**Comment by Malith-Rukshan** [comment]
Hey guys, your methods are working, but I’m facing an issue. When we control the DropDown trigger using the `isOpen` property, the [Keyboard Interactions](https://nextui.org/docs/components/dropdown#keyboard-interactions) (`Space` to open, `Esc` to close) and `shouldBlockScroll` don’t function correctly. Is there a way to open the DropDown on hover?
